.storybook/gu-auth/GuAuth.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect, useRef } from 'react';
+import dialogPolyfill from 'dialog-polyfill';
+import { css } from '@emotion/core';
+import { getLoginUrl } from '../utils';
+
+export const overlayStyles = css`
+    background-color: white;
+    border: 1px solid rgba(0, 0, 0, 0.1);
+    border-radius: 4px;
+    box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);
+    height: 200px;
+    width: 300px;
+    position: fixed;
+    top: 50%;
+    left: 50%;
+    margin-left: -150px;
+    margin-top: -100px;
+    z-index: 999;
+
+    ::backdrop {
+        position: fixed;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        background-color: rgba(0, 0, 0, 0.5);
+    }
+`;
+
+export const GuAuth = () => {
+    const [hasCheckedLoggedInStatus, setCheckedLoggedInStatus] = useState<boolean>(false);
+    const dialogRef = useRef<HTMLDialogElement | null>(null);
+
+    useEffect(() => {
+        if (dialogRef?.current) {
+            dialogPolyfill.registerDialog(dialogRef.current);
+        }
+    }, [dialogRef]);
+
+    useEffect(() => {
+        if (hasCheckedLoggedInStatus) {
+            dialogRef?.current?.close();
+        } else {
+            const alreadyOpen = dialogRef?.current?.open;
+
+            if (!alreadyOpen) {
+                dialogRef?.current?.showModal();
+
+                // Don't close the modal on escape
+                dialogRef?.current?.addEventListener('cancel', (e) => {
+                    e.preventDefault();
+                });
+            }
+        }
+    }, [hasCheckedLoggedInStatus]);
+
+    useEffect(() => {
+        const handleLogin = async () => {
+            const redirectToLogin = () => {
+                const currentLocation = window.location.href;
+
+                const newHref = `${getLoginUrl()}/login?returnUrl=${encodeURIComponent(
```

Comment:
The naming could be improve slightly here to sign post that it is a redirecting to a login

Reply:
👍  I updated the name of the const.